### ISSUE TITLE
Discussion of proposal guidelines and processes

### ISSUE THREAD

**Issue by JohnReppy** [SMLFamily/BasisLibrary#11]
This issue is for discussion of the draft guidelines for SML Basis Library proposals that have been posted to the [wiki](https://github.com/SMLFamily/BasisLibrary/wiki/Guidelines).

This issue is also intended as a forum for discussing the processes needed to submit, discuss, approve, and release changes to the Basis Library.

**Comment by MatthewFluet** [comment]
I think that the guidelines look good.

For sub-pages, perhaps use the naming scheme _YYYY-DDDA-NAME.ZZZ_, where _A_ is the sub-page alpha for the proposal (i.e., "a", "b", "c", ...).  Point being that it would be clear from the wiki page name whether it was a sub-page of a proposal or the root page of a proposal.

I don't quite like the fact that the entirety of a proposal is spread across the wiki, the git repository proper (for reference implementation), and GitHub issues, but don't see a better alternative.

**Comment by JohnReppy** [comment]
I agree about the naming scheme for subpages.  I'll update the guidelines.

W.r.t. spreading the content, that is really an artifact of building on github and trying to choose the best tool for each particular part.  We could move proposals into the repository, but we would lose the wiki support for organizing them.

**Comment by JohnReppy** [comment]
I have posted a draft description of the [Process document](https://github.com/SMLFamily/BasisLibrary/wiki/Process) to the wiki.  Please discuss.

**Comment by RobertHarper** [comment]
Looks like a good plan!

**Comment by MatthewFluet** [comment]
I think that the process document looks fine.  Perhaps update the template in `Code/examples` to include a Managing Editor field in the header.

**Comment by JohnReppy** [comment]
I've added a "Managing editor" field to the example.

**Comment by RobertHarper** [comment]
This is to record that I despise camel case and wish that we would change that.  It's not ML, it smacks of Smalltalk.  Ugh.

**Comment by MatthewFluet** [comment]
@RobertHarper Was `snake_case` the traditional ML convention?  Thinking of modern, module-rich MLs, I wondered if sufficient use of modules would obviate the need for complex identifiers (e.g., `TextIO.In.open`, rather than `TextIO.openIn`).

**Comment by RobertHarper** [comment]
yes, snake_case was the convention.  it’s true that proper use of modules avoids the need for openIn, etc.

bob

**Comment by JohnReppy** [comment]
snake_case was the standard for SML'90 (and before), but that changed for SML'97.

**Comment by RobertHarper** [comment]
not to my knowledge

**Comment by JohnReppy** [comment]
It did change, because we decided on a different standard for the Basis Library (Section 1.1.1 in the book).

**Comment by RobertHarper** [comment]
well, i don’t remember agreeing to that.  in any case my proposal for successor ml is to undo this.

bob

**Comment by JohnReppy** [comment]
I think that it depends on how much backward compatibility we want to preserve, but this discussion should probably be moved to a Successor ML issue, since this issue if to discussing the process for evolving the Standard ML Basis Library
